src/client/datascience/baseJupyterSession.ts

```diff
@@ -150,58 +155,15 @@ export abstract class BaseJupyterSession implements IJupyterSession {
         }
         return this.session.kernel?.requestKernelInfo();
     }
-    public async changeKernel(
-        resource: Resource,
-        kernelConnection: KernelConnectionMetadata,
-        timeoutMS: number
-    ): Promise<void> {
-        this.resource = resource;
-        let newSession: ISessionWithSocket | undefined;
-        // If we are already using this kernel in an active session just return back
-        const currentKernelSpec =
-            this.kernelConnectionMetadata && kernelConnectionMetadataHasKernelSpec(this.kernelConnectionMetadata)
-                ? this.kernelConnectionMetadata.kernelSpec
-                : undefined;
-        const kernelSpecToUse = kernelConnectionMetadataHasKernelSpec(kernelConnection)
-            ? kernelConnection.kernelSpec
-            : undefined;
-        if (this.session && currentKernelSpec && kernelSpecToUse && this.kernelConnectionMetadata) {
-            // If we have selected the same kernel connection, then nothing to do.
-            if (this.kernelConnectionMetadata.id === kernelConnection.id) {
-                traceInfoIfCI(`Kernels are the same, no switching necessary.`);
-                return;
-            }
-        }
-        trackKernelResourceInformation(resource, { kernelConnection });
-        newSession = await this.createNewKernelSession(resource, kernelConnection, timeoutMS);
-
-        // This is just like doing a restart, kill the old session (and the old restart session), and start new ones
-        if (this.session) {
-            this.shutdownSession(this.session, this.statusHandler, false).ignoreErrors();
-            this.restartSessionPromise?.then((r) => this.shutdownSession(r, undefined, true)).ignoreErrors(); // NOSONAR
-        }
-
-        traceInfoIfCI(`Switched notebook kernel to ${kernelSpecToUse?.display_name}`);
-
-        // Update our kernel connection metadata.
-        this.kernelConnectionMetadata = kernelConnection;
-
-        // Save the new session
-        this.setSession(newSession);
-
-        // Listen for session status changes
-        this.session?.statusChanged.connect(this.statusHandler); // NOSONAR
-    }
-
-    public async restart(timeout: number): Promise<void> {
+    public async restart(): Promise<void> {
```

Comment:
Removed the arguments this way the signature is identical to Jupyter Labs `Kernel.IKernelConnection.restart`